RiotSwiftUI/Modules/Room/UserSuggestion/Service/UserSuggestionService.swift

```diff
@@ -48,54 +48,71 @@ class UserSuggestionService: UserSuggestionServiceProtocol {
     private let roomMembersProvider: RoomMembersProviderProtocol
     
     private var suggestionItems: [UserSuggestionItemProtocol] = []
+    private let currentTextTriggerSubject = CurrentValueSubject<String?, Never>(nil)
+    private var cancellables = Set<AnyCancellable>()
     
     // MARK: Public
     
-    var items: CurrentValueSubject<[UserSuggestionItemProtocol], Never>
-    var currentTextTrigger: String?
+    var items = CurrentValueSubject<[UserSuggestionItemProtocol], Never>([])
+    
+    var currentTextTrigger: String? {
+        currentTextTriggerSubject.value
+    }
     
     // MARK: - Setup
     
     init(roomMembersProvider: RoomMembersProviderProtocol) {
         self.roomMembersProvider = roomMembersProvider
-        self.items = CurrentValueSubject([])
+        
+        currentTextTriggerSubject
+            .removeDuplicates()
+            .debounce(for: 0.5, scheduler: RunLoop.main)
```

Comment:
I think these should be the other way round, no? E.g. Quickly typing `s` `⌫` within the debounce time, then we would filter the duplicate so as to not trigger another fetch for the same string.